src/LinalgMsg.chpl

```diff
@@ -307,60 +305,53 @@ module LinalgMsg {
         for i in 0..<m do
             for j in 0..<n do
                 for l in 0..<k do
-                    C[i, j] += A[i, l] * B[l, j];
+            if t != bool {
+                        C[i, j] += A[i, l] * B[l, j];
+            } else {
+              C[i,j] |= A[i, l] & B[l, j];
+            }
     }
 
-    @arkouda.registerND
-    proc transposeMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-        if nd < 2 {
-            const errorMsg = "Matrix transpose with arrays of dimension < 2 is not supported";
-            linalgLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-            return new MsgTuple(errorMsg, MsgType.ERROR);
-        }
+    // Transpose an array.
 
-        const name = msgArgs.getValueOf("array"),
-              rname = st.nextName();
+    @arkouda.instantiateAndRegister
+    proc transpose(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
```

Comment:
Some of these could use the `registerCommand` annotation instead of `instantiateAndRegister` to simplify things further. I don't think it's necessary to change what you have here, just pointing it out for future reference.

```chapel
@arkouda.registerCommand
proc transpose(array: [?d] ?t): [] t
  where d.rank >= 2
{
  var outShape = array.shape;
  outShape[outShape.size-2] <=> outShape[outShape.size-1];
  var ret = makeDistArray(outShape, t);
  doTranspose(array, ret);
  return ret;
}
```

The argument parsing and symbol creation code will be generated automatically.

Reply:
I couldn't get this to work easily so I made a separate ticket: https://github.com/Bears-R-Us/arkouda/issues/3656